my-app/src/components/LoginContent/LoginContent.jsx

```diff
@@ -1,25 +1,104 @@
 import { Link } from 'react-router-dom';
 import './Login.css';
+import { useState } from 'react';
+import Input from './Input.jsx';
+import { isEmail, hasLetterAndNumber, hasMinLength } from './validation.js';
 
 export default function LoginContent() {
+  const [enteredValues, setEnteredValues] = useState({
+    email: '',
+    password: '',
+  });
+
+  const [notValidField, setNotValidField] = useState({
+    email: false,
+    password: false,
+  });
+
+  const [disabledButton, setDisabledButton] = useState(false);
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    console.log(enteredValues);
+    alert(JSON.stringify(enteredValues));
+  }
+
+  function handleInputChange(identifier, value) {
+    setEnteredValues(prev => ({
+      ...prev,
+      [identifier]: value,
+    }));
+    setNotValidField(prevValid => ({
+      ...prevValid,
+      [identifier]: false,
+    }));
+  }
+
+  const passwordIsValid =
+    hasLetterAndNumber(enteredValues.password) && hasMinLength(enteredValues.password);
+  const userNameIsValid = isEmail(enteredValues.email);
+
+  function handleBlur(identifier) {
+    if (identifier === 'email') {
+      if (!userNameIsValid) {
+        setNotValidField(prevValid => ({
+          ...prevValid,
+          [identifier]: true,
+        }));
+        setDisabledButton(true);
+      } else {
+        setDisabledButton(false);
+      }
+    }
+
+    if (identifier === 'password') {
+      if (!passwordIsValid) {
+        setNotValidField(prevValid => ({
+          ...prevValid,
+          [identifier]: true,
+        }));
+        setDisabledButton(true);
+      } else {
+        setDisabledButton(false);
+      }
```

Comment:
These two ifs are very similar. You can get rid of the code duplication. For example:

```
    const validation = {
      email: isEmail,
      password: password => hasLetterAndNumber(password) && hasMinLength(password),
    };

    const isValid = validation[identifier](enteredValues[identifier]);
    setNotValidField(prevValid => ({
      ...prevValid,
      [identifier]: !isValid,
    }));
    setDisabledButton(!isValid);
```